Essentials/src/com/earth2me/essentials/commands/EssentialsCommand.java

```diff
@@ -150,17 +152,34 @@ private User getPlayer(final Server server, final User sourceUser, final String
                     return userMatch;
                 }
             }
+
             final User userMatch = ess.getUser(matches.get(0));
             if (getHidden || canInteractWith(sourceUser, userMatch)) {
                 return userMatch;
             }
         }
+
         throw new PlayerNotFoundException();
     }
 
     @Override
     public final void run(final Server server, final User user, final String commandLabel, final Command cmd, final String[] args) throws Exception {
         final Trade charge = new Trade(this.getName(), ess);
+
```

Comment:
If the player runs a *different* command, we should probably clear the pending command. We should also store the pending command with a timestamp and disregard it if it was too long ago, in case a player runs the command assuming they're going to see the confirmation and ends up accidentally running the command itself.

Reply:
Well, previously it was not right to use the timestamp to not decrease the server performance with a scheduler. Or do we solve it without a scheduler, so with the help of current and elapsed time?

Reply:
Also, the pending command will get removed when a player runs a different command, see 173 line.